dev-docs/RFCs/v8.0/binary-support-improvements-rfc.md

```diff
@@ -0,0 +1,186 @@
+
+* index-based accessor:
+
+    ```js
+    // binaryData is packed on the server, in the format of [x, y, r, g, b, a]
+    const DATA = {src: binaryData, length: binaryData.length / 6};
+
+    new ScatterplotLayer({
+      data: DATA,
+      getPosition: (object, {index, data, target}) => {
+        target[0] = data.src[index * 6];
+        target[1] = data.src[index * 6 + 1];
+        target[2] = 0;
+        return target;
+      },
+      getColor: (object, {index, data, target}) => {
+        target[0] = data.src[index * 6 + 2] * 255;
+        target[1] = data.src[index * 6 + 3] * 255;
+        target[2] = data.src[index * 6 + 4] * 255;
+        target[3] = data.src[index * 6 + 5] * 255;
+        return target;
+      },
+      ...
+    })
+    ```
+
+    This method involves writing an accessor that reads relevant information out of the binary blob at each object index. During attribute update, an internal buffer is allocated, and the accessor is called `numInstance` times to populate it. Hence the performance is comparable to using plain JavaScript array data at step 2 and 3.
+
+* supplying external buffers directly:
+
+    ```js
+    // binaryData is packed on the server, in the format of [x, y, r, g, b, a]
+    const DATA = {
+      length: binaryData.length / 6,
+      attributes: {
+        instancePositions: {value: binaryData, size: 2, stride: 4 * 6, offset: 0},
+        instanceColors: {value: binaryData, size: 4, stride: 4 * 6, offset: 4 * 2, normalized: true}
+      }
+    };
+
+    new PointCloudLayer({
+      data: DATA,
+      getNormal: [0, 0, 0]
+    });
+    ```
+
+    This method skips step 2 entirely, and is possibly the most performant we'll ever get. However, it requires the application (BE and FE) to have knowledge of the internal implementation of a layer, including attribute names, array types and layouts, which is not documented and prone to breakage between minor releases.
+
+
+## Goals
+
+Moving into v8.x, we want to make binary data a first-class citizen, in the following ways:
+
+* All core layers accept a variety of binary data inputs, without exposing the internal layer implementation
+* Binary data are directly uploaded to the GPU if possible, and basic packing operations (e.g. position interleving, transform matrix construction) are performed on the GPU instead of CPU
+
+
+## Proposal
+
+Adding an additional override to the `accessor` prop type that is a string. `props.data[accessor]` should yield a loaders.gl-compatible "attribute descriptor" object.
+We shall explain this in the documentation as follows:
+
+A layer prop of type `accessor` can optionally be a string. If a string is provided, it is used as the key to query from `data` for a JavaScript object containing the following fields:
```

Comment:
Another comment is that I feel the `accessor` terminology is being used to refer to something more fundamental (essentially a "GPU column"), and at some point the usage starts to feel "inverted". I.e. the deck.gl accessor is a function that is used to calculate an attribute or GPU column, but now anything related to the underlying data is named after the function.

There is also the problem that the luma.gl "attribute descriptor" object referenced here is also called "accessor" in luma.gl. 

I'd be game for an ideation session on this. If we are going to change or fine-tune the nomenclature here v8 would probably be a good (or at least, less bad) time.